calico/felix/futils.py

```diff
@@ -630,3 +632,34 @@ def find_longest_prefix(strs):
             else:
                 longest_prefix = longest_prefix[:shared_len]
     return longest_prefix
+
+
+def usage_count(calico_version, hostname, cluster_guid, cluster_size, cluster_type):
+    """Reports the cluster's guid, size and version to projectcalico.org.
+    Logs out of date calico versions, to the standard log file.
+
+    :param calico_version: the calico version
+    :param hostname: the agent's hostname
+    :param cluster_guid: the unique cluster identifier
+    :param cluster_size: the number of felix instances
+    :cluster_type: the type of cluster
+    """
+    _log.info("usage_count calico_version=%s, hostname=%s, guid=%s, size=%s, cluster_type=%s", calico_version, hostname, cluster_guid, cluster_size, cluster_type)
+    try:
+        url = 'https://t736v4ek4g.execute-api.ap-northeast-1.amazonaws.com/prod/calicoVersionCheck'
+
+        urllib3.disable_warnings()
+        http = urllib3.PoolManager()
+        fields = {
+            'version': calico_version,
+            'hostname': hostname,
+            'guid': cluster_guid,
+            'size': cluster_size,
+            'cluster_type': cluster_type
+        }
+        retries = urllib3.util.retry.Retry(connect=5, read=5, redirect=5)
```

Comment:
If you want exponential backoff, you need to set a backoff_factor as well (see your link for documentation).  This will be limited to 120s at most anyway though (BACKOFF_MAX).

If the intention here is to avoid breaking lambda limits, then I think we'd be better served by using a larger jitter (thus spreading more evenly), and perhaps jittering before the first attempt.